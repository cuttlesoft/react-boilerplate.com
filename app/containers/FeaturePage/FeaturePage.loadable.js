import React from 'react'
import { IntlProvider } from 'react-intl'

import loadable from 'utils/loadable'
import { LoadingIndicator } from 'components/LoadingIndicator'

/**
 * Asynchronously loads the component for FeaturePage
 */
export default loadable(() => import('./FeaturePage'), {
  fallback: (
    <IntlProvider locale="en">
      <LoadingIndicator />
    </IntlProvider>
  ),
})
