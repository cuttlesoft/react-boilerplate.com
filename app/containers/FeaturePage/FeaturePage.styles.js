import styled from 'styled-components'

export const List = styled.ul`
  font-family: Georgia, Times, 'Times New Roman', serif;
  padding-left: 1.75em;
`

export const ListItem = styled.li`
  margin: 1em 0;
`
export const ListItemTitle = styled.p`
  font-weight: bold;
`
