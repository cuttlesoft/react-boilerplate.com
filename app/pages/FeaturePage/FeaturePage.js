import React from 'react'
import { Helmet } from 'react-helmet'
import { FormattedMessage } from 'react-intl'

// Components
import { Box } from 'components/Box'
import { Container } from 'components/Container'
import { Heading } from 'components/Heading'
import { Text } from 'components/Text'

// Messages
import messages from './FeaturePage.messages'

/*
 * FeaturePage
 *
 * List all the features
 */
export default function FeaturePage() {
  const items = [
    { title: messages.scaffoldingHeader, body: messages.scaffoldingMessage },
    { title: messages.feedbackHeader, body: messages.feedbackMessage },
    { title: messages.routingHeader, body: messages.routingMessage },
    { title: messages.networkHeader, body: messages.networkMessage },
    { title: messages.intlHeader, body: messages.intlMessage },
  ]

  return (
    <>
      <Helmet>
        <title>Feature Page</title>
        <meta name="description" content="Feature page of React.js Boilerplate application" />
      </Helmet>

      <Container>
        <Heading>
          <FormattedMessage {...messages.header} />
        </Heading>

        <Box padding={{ top: 'small' }}>
          {items.map(item => (
            <Box margin="small">
              <Text weight="bold">
                <FormattedMessage {...item.title} />
              </Text>

              <Text>
                <FormattedMessage {...item.body} />
              </Text>
            </Box>
          ))}
        </Box>
      </Container>
    </>
  )
}
