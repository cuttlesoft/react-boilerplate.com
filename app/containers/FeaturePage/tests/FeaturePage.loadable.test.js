import React from 'react'
import { waitForElement, render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import FeaturePage from '../FeaturePage.loadable'

/**
 * Test the FeaturePage
 */
test('should lazy load and match the snapshot', async () => {
  const { container } = render(
    <IntlProvider locale="en">
      <FeaturePage />
    </IntlProvider>,
  )

  const lazyComponent = await waitForElement(() => container)
  await expect(lazyComponent).toMatchSnapshot()
})
