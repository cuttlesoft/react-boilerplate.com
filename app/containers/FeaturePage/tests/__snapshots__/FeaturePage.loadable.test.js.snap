// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should lazy load and match the snapshot 1`] = `
<div>
  <div
    class="Wrapper-sc-12uw37d-0 cyDAiv"
  >
    <div
      class="Circle__CirclePrimitive-ww56dy-0 fUyYna"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 diOPOW"
      rotate="30"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 fUMtxj"
      rotate="60"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 lkIQiQ"
      rotate="90"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 hezFzS"
      rotate="120"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 iMQpMp"
      rotate="150"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 kboFDM"
      rotate="180"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 gdFpMO"
      rotate="210"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 gDjBjr"
      rotate="240"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 ihJMIo"
      rotate="270"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 eCUgy"
      rotate="300"
    />
    <div
      class="Circle__CirclePrimitive-ww56dy-0 cqJrGJ"
      rotate="330"
    />
  </div>
</div>
`;
